src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -257,15 +258,37 @@ public List<GeneratorFile> generateElement(GeneratableElement element, boolean f
 		// generate all source files
 		List<GeneratorTemplate> generatorTemplateList = getModElementGeneratorTemplatesList(element.getModElement(),
 				performFSTasks, element);
+		List<GeneratorTemplatesList> templateLists = getModElementListTemplates(element.getModElement(), performFSTasks,
+				element);
 		if (generatorTemplateList != null) {
 			for (GeneratorTemplate generatorTemplate : generatorTemplateList) {
 				String templateFileName = (String) ((Map<?, ?>) generatorTemplate.getTemplateData()).get("template");
 
 				Map<String, Object> dataModel = generatorTemplate.getDataModel();
 				extractVariables(generatorTemplate, dataModel);
 
-				String code = getTemplateGeneratorFromName("templates").generateElementFromTemplate(element,
-						templateFileName, dataModel, element.getAdditionalTemplateData());
+				String code = "";
+				if (generatorTemplate.isListTemplate() && templateLists != null) {
+					boolean listItemFound = false;
+					for (GeneratorTemplatesList gtl : templateLists) {
+						for (int i = 0; i < gtl.listData().size(); i++) {
+							GeneratorTemplate listTemplate = gtl.getCorrespondingListTemplate(

```

Comment:
You call getCorrespondingListTemplate by getting file from `generatorTemplate` to get `listTemplate` that is also GeneratorTemplate, I don't understand how this works

Reply:
1. We iterate over `listData` on each `GeneratorTemplatesList`;
2. We check if current `GeneratorTemplatesList` defines a template with the same path;
3. If found, we generate current template using appropriate list item;
4. If not found, we treat the template as a normal one.

Reply:
I think this can be implemented without having to call getCorrespondingListTemplate as this way you do many iterations. Check my comment below, maybe it is related to this or not, but I don't really like how entangled this is right now

Reply:
✔️ Fixed, waiting for you to enable tests on the PR back.
![image](https://user-images.githubusercontent.com/71347607/200559797-6d4cb282-f4f4-41cc-9c4f-d63dd6231ac4.png)